core/trino-spi/src/main/java/io/trino/spi/connector/FixedSplitSource.java

```diff
@@ -29,6 +29,16 @@
     private final Optional<List<Object>> tableExecuteSplitsInfo;
     private int offset;
 
+    public FixedSplitSource()
```

Comment:
Let's add a factory method `emptySplitSource` instead, since that will read better in callers.